intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -204,55 +397,117 @@ class Passenger:
 
 ```python
-from driver import Driver
+from .driver import Driver
 
 class Passenger:
 
     def __init__(self):
         print("I'm creating an instance of Passenger!")
         self.favorite_new_driver = Driver()
-
-scarlet = Passenger()
 ```
 
-Now produces this output, which doesn't have a `NameError`!
+This `import` tells Python to get the `Driver` identifier from the `driver` module located in the same folder as `passenger.py`. Running once more with `python3 main.py`, Scarlet gets the following output:
 
 ```
 I'm creating an instance of Passenger!
 I'm creating an instance of Driver!
 ```
 
-Again, the search for a matching module is relative to the file that is importing. In this example, the `driver` and `passenger` modules are in the same package, so there is no need to specify a package name. That's why `from driver` will suffice!
+No more `NameError` problems!
+
+Scarlet used an appropriate `import` statement for each case where Python needed to know where to find the identifier. From `main.py` she had to use a package full name import, since `main.py` is not part of the `ride_share_app` package. But from `passenger.py` she could use a relative `import`, since `passenger.py` and `driver.py` are in the same package.
+
+### !callout-info
+
+## `__main__.py` Is Another Special Package File
+
+We have already seen the special `__init__.py` file that Python uses to mark package folders. There is another special package file that can help us organize the way we start our programs.
+
+<br />
+
+If our package includes a file called `__main__.py`, we can start our programs like `python3 -m package_name`. We won't go into the details here, but follow your curiosity!
+
+### !end-callout
+
+### Don't Forget the Tests
+
+Scarlet hasn't written tests for her app yet. Take a moment to think about where she should add tests to this project, then reveal a possible structure below.
 
-Inside the matching `driver` module, it will load whatever is defined with the name `Driver`, in this case, a class. Inside of our `passenger` module, now we can call `Driver()` or any other `Driver` behavior `Driver`!
+<details style="max-width: 700px; margin: auto;">
+    <summary>
+      Click to reveal a possible project layout that includes tests.
+    </summary>
+
+```
+ride-share-app/
+├── README.md
+├── requirements.txt
+├── main.py
+├── ride_share_app
+│   ├── __init__.py
+│   ├── driver.py
+│   └── passenger.py
+└── tests
+    ├── __init__.py
+    ├── driver_test.py
+    └── passenger_test.py
+```
+
+Notice that the `tests` folder is in the project root folder (`ride-share-app`), not in the package folder (`ride_share_app`).
+
+</details>
 
 ## Debugging Imports
 
 Here are some common bugs that we may come across with importing:
 
+### `ModuleNotFoundError`
+
```

Comment:
I could imagine a small lead-up sentence:

```suggestion
### `ModuleNotFoundError`

Consider this stack trace caused from running `passenger.py`

```

Reply:
Ha ha.  Yeah. This struck me as a little abrupt too, just not enough to do anything about.

Thanks for the suggestions though!  I'll take them. 🙂